bin/upgrade_analyzer

```diff
@@ -0,0 +1,211 @@
+        comment << report.report
+      end
+
+      overall = reports.last
+
+      if overall.deprecation_warnings_changed?
+        github.add_labels_to_an_issue([UPGRADE_WARNING])
+      end
+
+      if overall.failed?
+        github.add_labels_to_an_issue([UPGRADE_REJECTED])
+      else
+        github.add_labels_to_an_issue([UPGRADE_ACCEPTED])
+      end
+
+      github.add_comment(comment.string)
+    end
+
+    def get_reports(build, results, base_results)
+      reports = []
+      overall_base_result = JobResult.new("Overall", description: base_branch(build))
+      overall_feature_result = JobResult.new("Overall", description: "current")
+
+      results.each_with_index do |result, index|
+        base_result = base_results[index]
+
+        overall_base_result << base_result
+        overall_feature_result << result
+
+        reports << ResultReporter.new(base_result, result)
+      end
+      reports << ResultReporter.new(overall_base_result, overall_feature_result)
+    end
+
+    def remove_labels(github)
+      github.remove_label(UPGRADE_ACCEPTED)
+      github.remove_label(UPGRADE_REJECTED)
+      github.remove_label(UPGRADE_REBASE)
+    end
+
+    def comparison_is_valid?(results, base_results)
+      results.length == base_results.length
+    end
+
+    def report_invalid_comparison(github)
+      error_message = "Cannot be compared to base branch. Manual comparison is necessary."
+      p error_message
+      github.add_comment("Upgrade Status: #{error_message}")
+      github.add_labels_to_an_issue([UPGRADE_REBASE])
+    end
+
+    def analyze_job(job, name)
+      p "Analyzing job: #{job.number}"
+      link = "<a href='https://travis-ci.com/#{repo_name}/jobs/#{job.id})'>#{name}</a>"
+      body = job.log.clean_body
+      match = body.match(/(?<tests>\d+) tests, (?<passed>\d+) passed, (?<failures>\d+) failures, (?<errors>\d+) errors/)
+      return unless match
+
+      summary = DeprecationSummary.new(StringIO.new(body).readlines)
+      JobResult.new(job.number,
+        description: link,
+        tests: match[:tests],
+        passed: match[:passed],
+        failures: match[:failures],
+        errors: match[:errors],
+        deprecations: summary.deprecations)
+    end
+
+    def repo
+      @repo ||= Travis::Pro::Repository.find(repo_name)
+    end
+  end
+end
+
+options = {}
+
+OptionParser.new do |opts|
+  opts.on("-b[BUILD_NUMBER]", "--build[=BUILD_NUMBER]", "The Travis build number to analyze. Ignored in listen mode.") do |build_number|
+    options[:build] = build_number
+  end
+  opts.on("-l", "--listen", "Listen for builds to complete, analyzing PR's as their builds finish") do |repo|
+    options[:listen] = repo
+  end
+  opts.on("-rREPOSITORY_NAME", "--repo=REPOSITORY_NAME", "GitHub repository to use. Include organization/user name and repo name") do |repo|
+    options[:repo] = repo
+  end
+  opts.on("-t[GITHUB_TOKEN]", "--token[=GITHUB_TOKEN]", "Your personal GitHub token") do |github_token|
```

Comment:
A bit lower in the file there is a fallback to using ENV var ( if it exist ) when command line switch is not provided.  Indeed for security using the ENV would be preferable.  Though, somehow I love that we support both options, leaving the choice to the user depending of her use case.